src/mame/taito/taito_f3_v.cpp

```diff
@@ -958,68 +995,74 @@ inline int taito_f3_state::pivot_inf::y_index(int y) const
 }
 
 template<typename Mix>
-bool taito_f3_state::mix_line(Mix &gfx, mix_pix *z, pri_mode *pri, const f3_line_inf &line, const clip_plane_inf &range)
+bool taito_f3_state::mix_line(const Mix &gfx, mix_pix &z, pri_mode &pri, const f3_line_inf &line, const clip_plane_inf &range)
 {
 	const int y = gfx.y_index(line.y);
 	const u16 *src = &gfx.bitmap.src->pix(y);
 	const u8 *flags = gfx.bitmap.flags ? &gfx.bitmap.flags->pix(y) : nullptr;
 
 	for (int x = range.l; x < range.r; x++) {
-		if (gfx.blend_mask() == pri[x].src_blendmode)
+		if (gfx.blend_mode == pri.src_blendmode[x])
 			continue; // note that layers cannot blend against the same blend mode
 
 		const int real_x = gfx.x_sample_enable ? mosaic(x, line.x_sample) : x;
 		const int gfx_x = gfx.x_index(real_x);
+
+		if constexpr (std::is_same_v<Mix, sprite_inf>) {
+			if (BIT(src[gfx_x], 10, 2) != gfx.index)
+				continue;
+		}
```

Comment:
Can you avoid using an `if constexpr` checking the type of the layer helper here?  It goes against the spirit of object oriented programming.

For example add to `struct mixable`:
```c++
	bool skip_pixel(u16 p) const { return false; }
```

Add to `struct sprite_inf`:
```c++
	bool skip_pixel(u16 p) const { return BIT(pix, 10, 2) != index; }
```

Then here you can avoid the messiness of comparing types:
```c++
		if (gfx.skip_pixel(src[gfx_x]))
			continue;
```

(Member function and parameter names are just for example, and there may be better choices.)